packages/kcmsf/src/pages/match.tsx

```diff
@@ -10,27 +10,90 @@ import { Judge } from "../utils/match/judge";
 import { expiryTimestamp, parseSeconds } from "../utils/time";
 
 type TimerState = "Initial" | "Started" | "Finished";
-
+type TeamFetchRes = {
+  id: string;
+  name: string;
+  entryId: string;
+  members: string[2];
+  clubName: string;
+  robotType: RobotType;
+  category: "elementary" | "open";
+  isEntered: boolean;
+};
 export const Match = () => {
-  const matchInfo = useLocation().state as MatchInfo | undefined;
-  const isExhibition = matchInfo == null;
+  const { id, matchType } = useParams<{ id: string; matchType: MatchType }>();
+  const isExhibition = id == null || matchType == null;
+  const [matchInfo, setMatchInfo] = useState<MatchInfo | undefined>(undefined);
+  const [matchJudge, setMatchJudge] = useState<Judge>(
+    new Judge(
+      { multiWalk: false },
+      { multiWalk: false },
+      { matchInfo },
+      { matchInfo }
+    )
+  );
+  useEffect(() => {
+    const fetchMatchInfo = async () => {
+      const res = await fetch(
+        `${import.meta.env.VITE_API_URL}/match/${matchType}/${id}`,
+        { method: "GET" }
+      );
+      const data = (await res.json()) as {
+        id: string;
+        matchType: MatchType;
+        left: { id: string; teamName: string };
+        right: { id: string; teamName: string };
+      };
+      const leftres = await fetch(
```

Comment:
```suggestion
      const leftRes = await fetch(
```
`leftdata` `rightres` `rightdata`も同様です。